web/src/utilities/utilities.js

```diff
@@ -611,6 +611,38 @@ export const Utilities = {
     }
   },
 
+  isInitialAppInstall(app) {
+    if (!app) {
+      return false;
+    }
+
+    if (app.downstream?.currentVersion) {
+      // a version has already been deployed
+      return false;
+    }
+
+    if (
+      !app.downstream?.pendingVersions ||
```

Comment:
this can be simplified:
```suggestion
      !app.downstream?.pendingVersions?.length
```